pkg/membership.go

```diff
@@ -0,0 +1,180 @@
+package pkg
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"graft/pkg/domain"
+	secondaryAdapter "graft/pkg/infrastructure/adapter/secondary"
+	secondaryPort "graft/pkg/infrastructure/port/secondary"
+)
+
+func AddClusterPeer(newPeer domain.Peer, clusterPeer domain.Peer) (chan struct{}, error) {
+	// 1. Get cluster leader
+	leaderNotFound := false
+	leader, err := getClusterLeader(clusterPeer)
+	switch err {
+	case nil:
+		break
+	case errLeaderNotFound:
+		leaderNotFound = true
+	default:
+		return nil, err
+	}
+
+	if leaderNotFound {
```

Comment:
This is a bit tricky, but  put simply:
since we are not able to get the Leader from the cluster configuration (either partition or election) we have a potentially stale configuration.

We can still try to start the node if the node host/id is contained in the configuration and active, because it would mean that the node is already known by the cluster. But not necessarily active.

This is better than nothing, but we should make clear that is is not ideal and the client should by himself that the node is indeed active in the entire cluster and not only on the node from which the configuration was fetched.